src/main/java/com/ifactory/press/db/solr/search/SafariBlockJoinQuery.java

```diff
@@ -290,16 +215,27 @@ public Explanation explain(int docBase) throws IOException {
             );
         }
 
-        @Override
+        // removed iterator so this may be needed somewhere else //TODO 
         public long cost() {
-            return childScorer.cost();
+            return childScorer.iterator().cost();
+        }
+
+        @Override  // rivey - iterator method added here
+        public SafariDocIdSetIterator iterator() {
+            return safDocSetIterator;
         }
 
     }
 
-    @Override
-    public void extractTerms(Set<Term> terms) {
-        childQuery.extractTerms(terms);
+    public void extractTerms(Set<Term> set) {  // rivey  Does this appear somewhere else? was an override
+        IndexReader emptyReader;
+        try {
+            emptyReader = new MultiReader();
+            Set<Term> termSet = new HashSet<Term>();
+            new IndexSearcher(emptyReader).createNormalizedWeight(childQuery, false).extractTerms(termSet);
+        } catch (IOException ex) {
+            Logger.getLogger(SafariBlockJoinQuery.class.getName()).log(Level.SEVERE, null, ex);
+        }
```

Comment:
I feel like this is the third time I've seen this same chunk of logic. Is there anyway we can abstract it out somewhere for re-use?